frontend/packages/core-api-client/src/client/Form.ts

```diff
@@ -32,4 +34,24 @@ export class FormClient {
       200,
     );
   };
+
+  getAncestors = async (formId: string): Promise<FormDefinition[]> => {
+    const formResponse = await this.get({ id: formId });
+
+    if (!formResponse.response) {
+      throw new Error(formResponse.error);
+    }
+
+    const reference = formResponse.response.fields.find(
+      (field: FieldDefinition) => field.fieldType.reference,
+    );
+
+    if (reference && reference.fieldType.reference) {
+      const result = await this.getAncestors(
+        reference.fieldType.reference?.formId,
+      );
+      return [...result, formResponse.response];
+    }
```

Comment:
I think the behaviour we want is slightly different. Rather than getting every reference field, we just want to get the reference fields that are keys.